### PR TITLE
support Oci image

### PR DESCRIPTION
**Pull request by gao-feng**
most of codes are backport from https://github.com/docker/docker/pull/26369
```
#./hyperctl images
REPOSITORY          TAG                 IMAGE ID     
busybox            latest                7968321274dc    
centos              7                   980e0e4c79ec  

//save centos:7 image as refname "hello" and busybox:latest image as refname "world" in mix-oci.tar usging oci image format
#./hyperctl save -o /tmp/mix-oci.tar -f oci -r centos:7=hello -r busybox:latest=world centos:7 busybox:latest

// untar mix-oci.tar and find index.json file
# cat index.json |jq
{
  "schemaVersion": 2,
  "manifests": [
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:351c1c2957c21a76d7b74e94afabd6429d4a3d16cbe6ee7e7b1f351e39125490",
      "size": 346,
      "annotations": {
        "org.opencontainers.ref.name": "hello"
      },
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:87fe700a194b98ab383582f9981539d9a69d49e1143e91fe0e75c4a92f5de42e",
      "size": 344,
      "annotations": {
        "org.opencontainers.ref.name": "world"
      },
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    }
  ]
}

// rm centos:7 and busybox:latest images for loading them from mix-oci.tar
#./hyperctl rmi centos:7 busybox:latest

// load centos and busybox images from mix-oci.tar and retag them as centos:test and busybox:test
./hyperctl load -i /tmp/oci/mix-oci.tar -r centos:7=hello -r busybox:test=world

#./hyperctl images
REPOSITORY          TAG                 IMAGE ID     
busybox           test                7968321274dc    
centos              test                   980e0e4c79ec 
```

### CODE REVIEW

**Comment by gnawux** [comment]
looks ` vendor/github.com/docker/docker/` was changed, but not updated in `Godeps/Godeps.json`

**Comment by gao-feng** [comment]
@gnawux The changes are added by myself, we need to access dockerDaemon.layerstore imagestore in hyperd daemon, first way is maintaining the changed docker-daemon package in hyperd, second way is changing docker-daemon directly in vendor. Since we will reuse docker's codes when [pr](https://github.com/docker/docker/pull/26369) is merged, I chose the second way to resolve this problem it is simplest.

File: client/load.go [hunk 1]
```diff
@@ -5,12 +5,15 @@ import (
...
 
+	runconfigopts "github.com/docker/docker/runconfig/opts"
 	gflag "github.com/jessevdk/go-flags"
 )
 
 func (cli *HyperClient) HyperCmdLoad(args ...string) error {
 	var opts struct {
-		Input string `short:"i" long:"input" value-name:"\"\"" description:"Read from a tar archive file, instead of STDIN"`
+		Input string   `short:"i" long:"input" value-name:"\"\"" description:"Read from a tar archive file, instead of STDIN"`
+		Name  string   `short:"n" long:"name" value-name:"\"\"" description:"Name to use when loading OCI image layout tar archive"`
```
Comment (laijs) [inline client/load.go]: how about --oci ?

Comment (gao-feng) [inline client/load.go]: this option is used as the name of loaded image. and this option is consist with https://github.com/docker/docker/pull/26369

**Comment by laijs** [comment]
retest it please @hykins

**Comment by laijs** [comment]
retest this please @hykins

**Comment by gao-feng** [comment]
retest this please @hykins

**Pull request merged**